datumaro/plugins/data_formats/datumaro_binary/base.py

```diff
@@ -24,13 +24,22 @@ def __init__(self, path: str, encryption_key: Optional[bytes] = None):
         self._crypter = Crypter(encryption_key)
         super().__init__(path)
 
+    def _get_dm_format_version(self, path: str) -> str:
+        with open(path, "rb") as fp:
+            self._fp = fp
+            self._check_signature()
+            self._check_encryption_field()
+            dm_format_version = self._read_version()
```

Comment:
```suggestion
            self._fp = fp
            self._check_signature()
            dm_format_version = self._read_version()
            self._check_encryption_field()
```
Please check the version first before checking the encryption field.